services/src/main/java/org/exoplatform/officeonline/rest/WOPIResource.java

```diff
@@ -117,6 +125,127 @@ public WOPIResource(WOPIService wopiService) {
+                         .entity("{\"error\": \"" + e.getMessage() + "\"}")
+                         .type(MediaType.APPLICATION_JSON)
+                         .build();
+        } catch (RepositoryException e) {
+          return Response.status(Status.INTERNAL_SERVER_ERROR)
+                         .entity("{\"error\": \"Internal error while saving content\"}")
+                         .type(MediaType.APPLICATION_JSON)
+                         .build();
+        } catch (IOException e) {
+          return Response.status(Status.INTERNAL_SERVER_ERROR)
+                         .entity("{\"error\": \"Cannot get request body\"}")
+                         .type(MediaType.APPLICATION_JSON)
+                         .build();
+        }
+      } else {
+        return Response.status(Status.UNAUTHORIZED)
+                       .entity("{\"error\": \"Couldn't build config from access token\"}")
+                       .type(MediaType.APPLICATION_JSON)
+                       .build();
+      }
+    } else {
+      return Response.status(Status.BAD_REQUEST)
+                     .entity("{\"error\": \"Wrong operation\"}")
+                     .type(MediaType.APPLICATION_JSON)
+                     .build();
+    }
+
+  }
+
+  /**
+   * Files.
+   *
+   * @param uriInfo the uri info
+   * @param request the request
+   * @param operation the operation
+   * @param fileId the file id
+   * @return the response
+   */
+  @GET
+  @Path("/files/{fileId}/contents")
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response getFile(@Context UriInfo uriInfo,
+                          @Context HttpServletRequest request,
+                          @Context ServletContext context,
+                          @PathParam("fileId") String fileId) {
+
+    verifyProofKey(request);
+    EditorConfig config = (EditorConfig) context.getAttribute(EDITOR_CONFIG_PARAM);
+    if (config != null) {
+      try {
+        DocumentContent content = wopiService.getContent(fileId, config);
+        return Response.ok().header(ITEM_VERSION, content.getVersion()).entity(content.getData()).type(content.getType()).build();
+      } catch (BadParameterException e) {
+        return Response.status(Status.BAD_REQUEST)
+                       .entity("{\"error\": \"" + e.getMessage() + "\"}")
+                       .type(MediaType.APPLICATION_JSON)
+                       .build();
+      } catch (FileNotFoundException e) {
+        return Response.status(Status.NOT_FOUND)
+                       .entity("{\"error\": \"" + e.getMessage() + "\"}")
+                       .type(MediaType.APPLICATION_JSON)
+                       .build();
+      } catch (OfficeOnlineException e) {
+        return Response.status(Status.BAD_REQUEST)
+                       .entity("{\"error\": \"" + e.getMessage() + "\"}")
+                       .type(MediaType.APPLICATION_JSON)
+                       .build();
+      }
+
+    } else {
+      return Response.status(Status.UNAUTHORIZED)
```

Comment:
Is it true to respond with UNAUTHORIZED if config not found? Why not NOT_FOUND? Otherwise filter logic should add an extra flag to make us sure here it's auth problem (not something else, like missing filter or error in it that lead to missing attribute).

Reply:
We return UNAUTHORIZED because the config is built from an access token. 
According to WOPI docs we have to return 401 status code in case of the invalid access token.
I'll add the extra flag for the auth failure.